### ISSUE TITLE
Contact Objects

### ISSUE THREAD

**Issue by Bill-Irvine** [digitalocean/netbox#1344]
### Issue type: Feature Request

This may well be out of scope for Netbox as a technical application but thought i'd gauge interest in this feature.

I find myself managing several sites with the same contact. When that contact for example changes an email address, phone number, or someone takes over their role i am updating all the sites with that contact listed.

If this is changed to a contact object that we then attach to these sites we have a central place to manage these people without duplicating the data in separate sites.

From here we could add functionality such as adding several contacts to a site (primary, secondary), we could stipulate between a business contact or a technical contact, add which company they work for, have an api get request that lists contacts for a child object such as devices (useful for managing outages and notifications).

Let me know your thoughts.

**Comment by modir** [comment]
Not only would it be useful to have several contacts for sites, but for providers as well. There we often have as well a technical contact and a business contact.

**Comment by tb-killa** [comment]
We need this type of Informations too.
Please implement.

**Comment by SzymonKurowski** [comment]
I have similar business needs in my organization (multiple contacts per site). I like the approach described above.

Is there anyone who is currently working on that? I would like to offer my contribution.

**Comment by SzymonKurowski** [comment]
There is a one key point to discuss. How to deal with current contact information added to Site model.

To implement this feature it is necessary to add new model (i.e. Contact) and then if we move contact from Site to the Contact model – it will break a backward-compatibility. 

For the first iteration, I would leave Site contact fields (it can be treat as a main contact to the site) and add Contact model for any specific contacts related to the site. Additionally, each contact from Contact model can refer to more than one site.

Please let me know if you have any thoughts.

**Comment by mtbutler07** [comment]
+1 For adding multiple sources for contact information for sites.

**Comment by ryanmerolle** [comment]
I say for backwards compatibility, I would implement the way in which @SzymonKurowski mentions with one alteration:
- If no legacy site contact is recorded for a site, then netbox can default to showing the new contact fields.
- If a legacy site contact is recorded for a site, netbox will show the contact with a note directing the user to the new model.

**Comment by ryanmerolle** [comment]
It would also be useful to apply the contact object relationship to:

- Sites
- Regions
- Tenants
- Manufacturers
- Providers

You could argue for contacts to be linked to devices, racks, rack groups, and etc, but that would shift this tool to more of a CMDB for everything, and I am not sure we want to do that.

**Comment by adamkf** [comment]
@ryanmerolle: I agree this relationship might be useful across multiple object relationships.  We managed a couple thousand internal subnets in our org and currently use custom fields to assign ownership responsibility for given subnets (eg: Network Team, Dev Team 1, Dev Team 88, Individual Person 23, etc) so when our security group reaches out and says "who owns this infected machine" we have a starting point.

This FR has been around since 2017. Has anyone made progress toward it?  I'm happy to throw my time at as we have a need internally.

**Comment by athompson-merlin** [comment]
FWIW, I don't need extended contact capabilities on Sites, but I do desperately need Contacts on *Tenants* instead.
Having an optional Contact[s] object attachable to every top-level object would be ideal, who knows where I might need to hang contact information someday.

**Comment by jorp** [comment]
Would it be too much of a hassle to be able to bring contact information in from LDAP? Setting a contact as a specific user or group would be pretty beneficial.

**Comment by pisaniej** [comment]
What kind of relationship would a Contact have to an object?

For example, would `virtualization.VirtualMachine` be able to have multiple Contacts? I could see this being useful for an internal contact relating to an object, along with adding another for external vendor information.

**Comment by DanSheps** [comment]
And here I was thinking I was going to take care of contact objects.  Guess you beat me too it...

**Comment by pisaniej** [comment]
@jeremystretch, in theory, it should be possible to import a group in LDAP as a group in Netbox, along with it's members from LDAP as individual contact objects within Netbox, no?
